### ISSUE TITLE
getIn (prop keypaths)

### ISSUE THREAD

**Issue by tolmasky** [ramda/ramda#637]
Facebook's immutable has a nifty getIn method for grabbing a value at a key path:

```javascript
myMap.getIn(["a", "b", "c"], defaultValue)
```

I find myself piping together lots of prop calls to accomplish something similar:

```javascript
R.map(R.pipe(R.prop("id"), R.prop("name")), vars)
```

I made for myself something like getIn:

```javascript
R.map(R.getIn(["id", "name"]), vars)
```

If this is something others could want I could make a pull request.

**Comment by CrossEye** [comment]
@tolmasky Do you want to create a PR for this?

**Comment by tolmasky** [comment]
Hi yeah sorry, I need to split this out from my stuff and submit it. I'll
attempt do it this weekend

On Thursday, January 8, 2015, Scott Sauyet <notifications@github.com> wrote:

> @tolmasky <https://github.com/tolmasky> Do you want to create a PR for
> this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ramda/ramda/issues/637#issuecomment-69286299>.
>


-- 
Francisco Tolmasky
www.tolmasky.com
tolmasky@gmail.com

**Comment by tolmasky** [comment]
Just providing an update, this touched a couple other functions so I'm just updating all the tests.

**Comment by CrossEye** [comment]
Thanks for the update.

I look forward to seeing it (even if a little sad about losing the old API.)

**Comment by CrossEye** [comment]
Moving this to 0.10.0.  I don't think anyone is desperately waiting on it, and I think we really want to publish 0.9.0.

**Comment by jethrolarson** [comment]
I like the string sugar. I can see both approaches having their use though.
I suppose if you implement the array then the string version can just be `dotPath = compose(path, split('.'))`

**Comment by tolmasky** [comment]
Sorry, know this is taking me forever. But another thing that comes to mind
is that there seems to be no reason to not just support both:

getin(path)
{
if (typeof path === string) return getin(split(“.”))
else …
}

Seems then it works the way everyone expects.


On Fri, Feb 6, 2015 at 3:46 PM, Jethro Larson <notifications@github.com>
wrote:

> I like the string sugar. I can see both approaches having their use though.
> I suppose if you implement the array then the string version can just be dotPath
> = compose(path, split('.'))
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ramda/ramda/issues/637#issuecomment-73334325>.
>



-- 
Francisco Tolmasky
www.tolmasky.com
tolmasky@gmail.com

**Comment by buzzdecafe** [comment]
:-1: on type-checking the arguments in the function. no reason the user can't `path(split('.', str), obj)`.

**Comment by CrossEye** [comment]
I like the string sugar too.  I actually prefer it, even though I know it's not as general.  But I definitely wouldn't want the type-checking version.  We're trying to avoid that as much as we can in Ramda.  As @jethrolarson said, it's easy enough to compose, or add your own gloss atop the list version.

**Comment by AutoSponge** [comment]
`p = path.split ? path.split('.') : path;`
It's not exactly type-checking.

**Comment by jethrolarson** [comment]
I think the point is not about implementation but api. Ramda functions
don't take variable type inputs. Except for R.add, but that's probably an
oversight.

**Comment by CrossEye** [comment]
> I think the point is not about implementation but api.

Exactly.  It's about what's _simple_ as opposed to what's _easy_.  (And `add` just happens to fall out of the implementation; we don't advertise that it works on strings as well.)

**Comment by CrossEye** [comment]
@tolmasky ping
